src/TypeSpec/__Private/UntypedArraySpec.hack

```diff
@@ -39,7 +39,7 @@ final class UntypedArraySpec extends TypeSpec<array> {
         IncorrectTypeException::withValue($this->getTrace(), 'array', $value);
     }
 
-    $out = [];
+    $out = darray[];
     foreach ($value as $k => $v) {
       $out[$k] = $v;
     }
```

Comment:
Here we should probably just `return $value;`

IMHO `assert` should not change the type of the value so we shouldn't be converting arbitrary arrays to darray.